azurerm/internal/services/storage/storage_account_resource_test.go

```diff
@@ -36,864 +39,694 @@ func TestValidateArmStorageAccountName(t *testing.T) {
-		conn := acceptance.AzureProvider.Meta().(*clients.Client).Storage.AccountsClient
-
-		// Ensure we have enough information in state to look up in API
-		rs, ok := s.RootModule().Resources[resourceName]
-		if !ok {
-			return fmt.Errorf("Not found: %s", resourceName)
-		}
-
-		storageAccount := rs.Primary.Attributes["name"]
-		resourceGroup := rs.Primary.Attributes["resource_group_name"]
-
-		resp, err := conn.GetProperties(ctx, resourceGroup, storageAccount, "")
-		if err != nil {
-			return fmt.Errorf("Bad: Get on storageServiceClient: %+v", err)
-		}
-
-		if resp.StatusCode == http.StatusNotFound {
-			return fmt.Errorf("Bad: StorageAccount %q (resource group: %q) does not exist", storageAccount, resourceGroup)
-		}
-
-		return nil
+func (r StorageAccountResource) Exists(ctx context.Context, client *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := parse.StorageAccountID(state.ID)
+	if err != nil {
+		return nil, err
 	}
-}
-
-func testCheckAzureRMStorageAccountDisappears(resourceName string) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		// Ensure resource group exists in API
-		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
-		conn := acceptance.AzureProvider.Meta().(*clients.Client).Storage.AccountsClient
-
-		// Ensure we have enough information in state to look up in API
-		rs, ok := s.RootModule().Resources[resourceName]
-		if !ok {
-			return fmt.Errorf("Not found: %s", resourceName)
+	resp, err := client.Storage.AccountsClient.GetProperties(ctx, id.ResourceGroup, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
 		}
-
-		storageAccount := rs.Primary.Attributes["name"]
-		resourceGroup := rs.Primary.Attributes["resource_group_name"]
-
-		if _, err := conn.Delete(ctx, resourceGroup, storageAccount); err != nil {
-			return fmt.Errorf("Bad: Delete on storageServiceClient: %+v", err)
-		}
-
-		return nil
+		return nil, fmt.Errorf("retrieving Storage Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
+	return utils.Bool(true), nil
 }
 
-func testCheckAzureRMStorageAccountDestroy(s *terraform.State) error {
-	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
-	conn := acceptance.AzureProvider.Meta().(*clients.Client).Storage.AccountsClient
-
-	for _, rs := range s.RootModule().Resources {
-		if rs.Type != "azurerm_storage_account" {
-			continue
-		}
-
-		name := rs.Primary.Attributes["name"]
-		resourceGroup := rs.Primary.Attributes["resource_group_name"]
-
-		resp, err := conn.GetProperties(ctx, resourceGroup, name, "")
-		if err != nil {
-			return nil
-		}
-
-		if resp.StatusCode != http.StatusNotFound {
-			return fmt.Errorf("Storage Account still exists:\n%#v", resp.AccountProperties)
-		}
+func (r StorageAccountResource) Destroy(ctx context.Context, client *clients.Client, state *terraform.InstanceState) (*bool, error) {
```

Comment:
I don't think we need destroy functions? the new test shim framework uses the exists function to check that

Reply:
I implement this function because the storage_account resource has a `TestAccStorageAccount_disappears` test which requires a Destroy function